### PR TITLE
PARQUET-610: Print additional ColumnMetaData for each RowGroup

### PR DESCRIPTION
**Pull request by majetideepak**
Also fixes few warnings in release mode build

### CODE REVIEW

File: example/decode_benchmark.cc [hunk 1]
```diff
@@ -372,7 +372,7 @@ void TestDeltaLengthByteArray() {
       encoder.plain_encoded_len(), len);
   decoder.SetData(encoder.num_values(), buffer, len);
   for (int i = 0; i < encoder.num_values(); ++i) {
-    ByteArray v;
+    ByteArray v = {0, NULL};
```
Comment (majetideepak) [inline example/decode_benchmark.cc]: Few uninitialized variables caused warnings in release mode build.

File: src/parquet/types.cc [hunk 2]
```diff
@@ -49,17 +49,11 @@ std::string FormatValue(Type::type parquet_type, const char* val, int length) {
       break;
     }
     case Type::BYTE_ARRAY: {
-      const ByteArray* a = reinterpret_cast<const ByteArray*>(val);
-      for (int i = 0; i < static_cast<int>(a->len); i++) {
-        result << a[0].ptr[i] << " ";
-      }
+      result << val << " ";
```
Comment (majetideepak) [inline src/parquet/types.cc]: Min, Max statistics for BA and FLBA types have only the `bytes` and no `length` associated with them.

**Comment by majetideepak** [comment]
Sample output
Total rows: 2
Column 0: id (INT32)
Column 1: bool_col (BOOLEAN)
--- Row Group 0 ---
Column 0
  rows: 2, values: 2, null values: 0, distinct values: 0
  Statistics Not Set
  compression: UNCOMPRESSED, encodings: RLE PLAIN_DICTIONARY PLAIN 
  uncompressed size: 47, compressed size: 47
Column 1
  rows: 2, values: 2, null values: 0, distinct values: 0
  Statistics Not Set
  compression: UNCOMPRESSED, encodings: RLE PLAIN_DICTIONARY PLAIN 
  uncompressed size: 24, compressed size: 24

File statistics:
Total rows: 12000
Column 0: int_conseq (INT32)
Column 1: int_random (INT32)
--- Row Group 0 ---
Column 0
  rows: 12000, values: 12000, null values: 1000, distinct values: 0
  max: 5999, min: -5000
  compression: GZIP, encodings: PLAIN RLE BIT_PACKED 
  uncompressed size: 44051, compressed size: 15379
Column 1
  rows: 12000, values: 12000, null values: 2000, distinct values: 0
  max: 14938, min: -4866
  compression: GZIP, encodings: PLAIN RLE BIT_PACKED 
  uncompressed size: 40051, compressed size: 24214

**Comment by wesm** [comment]
Cool, I will merge this for now -- if we could, let's hold off on doing any more work exposing metadata in the public API, I wanted to work through this in PARQUET-573 (basically, a public API for interacting with the Thrift FileMetaData struct) so I will do some refactoring as a part of that

**Comment by wesm** [comment]
+1

**Comment by majetideepak** [comment]
I missed PARQUET-573. I agree, a separate public API will make it look cleaner. Thank you!

**Pull request closed without merging**